kf_lib_data_ingest/common/misc.py

```diff
@@ -124,19 +124,32 @@ def numeric_to_str(val, replace_na=False, na=None):
 
     val = str(val).strip()
     if val != '':
-        # Don't automatically change anything with leading zeros, scientific
-        # notation, or underscores (I don't care what PEP 515 says).
-        if (val[0] == '0') or (not re.fullmatch(r'[\d.]+', val)):
-            return val
+        # Try downcasting val
+        i_val = None
+        f_val = None
         try:
             f_val = float(val)
             i_val = int(f_val)
-            if i_val == f_val:
-                return str(i_val)
-        except Exception:
+        except ValueError:
             pass
+
+        # Don't automatically change anything with leading zeros
+        # (except something that equates to int 0), scientific
+        # notation, or underscores (I don't care what PEP 515 says).
+        if (
+            not (i_val == 0) and  # does not equate to int 0
+                (val[0] == '0') or  # starts with leading zeros
+                (not re.fullmatch(r'[\d.]+', val))  # scientific notation
+        ):
```

Comment:
The operator precedence is confusing on this conditional. Can you put parentheses in to clarify desired order of operations?

Reply:
Haha oops my bad, thats terrible